src/Utilities/MoistThermodynamics/src/MoistThermodynamics.jl

```diff
@@ -463,48 +739,244 @@ function saturation_adjustment(e_int, ρ, q_tot, T_init = T_triple)
+given a thermodynamic state `ts`.
+"""
+liquid_ice_pottemp(ts::ThermodynamicState) =
+  liquid_ice_pottemp(air_temperature(ts), air_pressure(ts), ts.q_tot,
+                     phase_partitioning_eq(ts)...)
+
+"""
+    mix_ratio_con(q_tot, q_liq, q_ice)
+
+The mixing ratio of condensed liquid
+to dry air where
+ - `q_tot` total specific humidity
+ - `q_liq` liquid specific humidity
+ - `q_ice` ice specific humidity
+"""
+mix_ratio_con(q_tot::DT, q_liq::DT, q_ice::DT) where {DT} =
+  (q_liq+q_ice)/(1-q_tot)
+
+"""
+    mix_ratio_con(ts::ThermodynamicState)
+
+The mixing ratio of condensed liquid
+to dry air given a thermodynamic state `ts`.
+"""
+mix_ratio_con(ts::ThermodynamicState) =
+  mix_ratio_con(ts.q_tot, phase_partitioning_eq(ts)...)
 
+
+"""
+    mix_ratio_vap(q_tot, q_liq, q_ice)
+
+The mixing ratio of liquid vapor
+to dry air where
+ - `q_tot` total specific humidity
+ - `q_liq` liquid specific humidity
+ - `q_ice` ice specific humidity
+"""
+mix_ratio_vap(q_tot::DT, q_liq::DT, q_ice::DT) where {DT} =
+  (q_tot-q_liq-q_ice)/(1-q_tot)
+
+"""
+    mix_ratio_vap(ts::ThermodynamicState)
+    mix_ratio_vap(q_tot, q_liq, q_ice)
+
+The mixing ratio of liquid vapor to dry air given a thermodynamic state `ts`.
+"""
+mix_ratio_vap(ts::ThermodynamicState) =
+  mix_ratio_con(ts.q_tot, phase_partitioning_eq(ts)...)
+
+"""
+    dry_pottemp(T, p, q_tot, q_liq, q_ice)
+
+The dry potential temperature where
+
+ - `T` temperature
+ - `p` pressure
+and, optionally,
+ - `q_tot` total specific humidity
+ - `q_liq` liquid specific humidity
+ - `q_ice` ice specific humidity
+ """
+dry_pottemp(T::DT, p::DT, q_tot::DT=DT(0), q_liq::DT=DT(0), q_ice::DT=DT(0)) where {DT} =
+  T / exner(p, q_tot, q_liq, q_ice)
+
+"""
+    dry_pottemp(ts::ThermodynamicState)
+
+The dry potential temperature, given a thermodynamic state `ts`.
+"""
+dry_pottemp(ts::ThermodynamicState) =
+  dry_pottemp(air_temperature(ts), air_pressure(ts), ts.q_tot,
+              phase_partitioning_eq(ts)...)
+
+
+"""
+    virtual_pottemp(T, p, q_tot, q_liq, q_ice)
+
+The virtual potential temperature where
+
+ - `T` temperature
+and, optionally,
+ - `q_tot` total specific humidity
+ - `q_liq` liquid specific humidity
+ - `q_ice` ice specific humidity
+"""
+virtual_pottemp(T::DT, q_tot::DT=DT(0), q_liq::DT=DT(0), q_ice::DT=DT(0)) where {DT} =
+  gas_constant_air(q_tot, q_liq, q_ice)/R_d*T
+
+"""
+    virtual_pottemp(ts::ThermodynamicState)
+
+The virtual potential temperature,
+given a thermodynamic state `ts`.
+"""
+virtual_pottemp(ts::ThermodynamicState) =
+  virtual_pottemp(air_temperature(ts), ts.q_tot, phase_partitioning_eq(ts)...)
+
+
+"""
+    liquid_pottemp(T, p, q_tot, q_liq, q_ice)
+
+The liquid potential temperature where
+
+ - `T` temperature
+ - `p` pressure
+and, optionally,
+ - `q_tot` total specific humidity
+ - `q_liq` liquid specific humidity
+ - `q_ice` ice specific humidity
+"""
+function liquid_pottemp(T::DT, p::DT, q_tot::DT=DT(0), q_liq::DT=DT(0), q_ice::DT=DT(0)) where {DT}
```

Comment:
Replace this by something like `liquid_ice_pottemp_sat`, and use `saturation_excess`, `phase_partitioning_eq` and `liquid_ice_pottemp` to compute it.